src/app/core/services/data.service.ts

```diff
@@ -203,22 +206,22 @@ export class DataService {
 
   /**
    * Converts the input csv information to an array of
-   * objects syncrhonously. Each object's values are
+   * parsed user data. Each object's values are
    * marked by the respective csv table headers.
    * @param csvText - csv information.
    * @return - Subjects that tracks the parsed data.
    */
-  private csvParser(csvText: string): {}[] {
+  private parseUsersData(csvText: string): ParsedUserData[] {
     const lines = csvText.split('\n').filter(v => v.trim());
     const headers = lines[0].split(',').map(h => h.trim());
-    const result = [];
+    const result: ParsedUserData[] = [];
     for (let i = 1; i < lines.length; i++) {
       const line = lines[i].trim();
       if (!line) {
         continue;
       }
```

Comment:
Not related to PR but
Given that we have already filtered empty lines with line 215, in what case will this condition be true?
Seems like we can remove this check

Reply:
True, let me look into this to see if it can be removed

Reply:
I've checked and removed. it It is not executed while using our current csv file